CHANGELOG.md

```diff
@@ -1,7 +1,252 @@
+- Added `alpaka::is{Accelerator,Device,Platform,Queue}` template variables #1818
+- Added accelerator tags which allow for accelerator-specific code paths without enabling the corresponding back-end #1804 #1814
+- Added support for `std::mdspan` #1788 #2048 #2052 #2053
+- Added `alpaka::ViewConst` which wraps another view but only allows non-modifying accesses #1746
+- `alpaka::{memcpy,memset}` now support temporary destination views #1743
+- Host memory alignment can now be specified by using the `ALPAKA_DEFAULT_HOST_MEMORY_ALIGNMENT` macro #1686
+- Added `alpaka::allocMappedBuf` for allocating device-accessible pinned host memory #1685 #1782
+  - Added related trait `alpaka::trait::hasMappedBufSupport` to query the host CPU for device-accessible pinned memory support #1782
+  - Added related utility function `alpaka::allocMappedBufIfSupported` to allocate device-accessible pinned memory if supported and regular memory otherwise #1782
+- Relocatable device code can now be enabled using the `alpaka_RELOCATABLE_DEVICE_CODE` CMake option #1467
+
+### Changed
+
+- API changes:
+  - **Breaking change**: alpaka platforms have been renamed from `alpaka::Pltf*` to `alpaka::Platform*` #2024 #2032
+  - **Breaking change**: alpaka platforms are now full objects instead of types #1988 #2051
+  - `operator<<(std::ostream&, WorkDivMembers const&)` is now a `friend` of `alpaka::WorkDivMembers` instead of a method #1829
+  - **Potentially breaking change**: Switched several view-related methods from `ALPAKA_FN_HOST_ACC` to `ALPAKA_FN_HOST` #1826
+  - Accelerators' copy/move constructors and assignment operators are now explicitly `delete`d #1825
+  - `alpaka::test::allocAsyncBufIfSupported` was moved into the general `namespace alpaka` #1782
+  - Removed unnecessary attribute `ALPAKA_FN_HOST_ACC` from defaulted functions #1761
+  - The `UniformCudaHip` types are now templated on traits-like `struct`s which encapsulate the CUDA or HIP API #1665
+- General behavioural changes:
+  - Simplified internals of `alpaka_add_{executable,library}` #2072 #2082
+  - **Breaking change**: Removed dummy atomics from memory fence implementations. Users now need to guarantee correctness themselves #2071
+  - In debug mode MSVC will use the `/Od` optimization level #1977
+  - In debug mode clang-based compilers will explicitly use the `-O0` optimization level #1977
+  - In debug mode `g++` will use the `-Og` optimization level #1977
+  - `-Werror` and its MSVC equivalent `/WX` are no longer enabled by default when `BUILD_TESTING` is set to `ON` #1977
+  - A platform's internal `std::vector` containing the `alpaka::Device`s now reserves the necessary memory before initialization #1926
+  - **Potentially breaking change**: `ALPAKA_FN_INLINE` now enforces inlining for platforms other than CUDA and HIP #1918
+  - Replaced `alpaka::core::ConcurrentExecPool` with `alpaka::core::CallbackThread` in all queue implementations #1870
```

Comment:
```suggestion
  - Replaced `alpaka::core::ConcurrentExecPool` with `alpaka::core::CallbackThread` in all queue implementations #1870
```
That has "breaking" behavior as well, since tasks enqueued by the users are now guaranteed to be finished, and the queue dtor blocks for that.